src/main/kotlin/org/rust/cargo/project/model/impl/CargoSyncTask.kt

```diff
@@ -201,6 +208,14 @@ private fun fetchCargoWorkspace(context: CargoSyncTask.SyncContext): TaskResult<
         val cargo = toolchain.cargoOrWrapper(projectDirectory)
         try {
             val projectDescriptionData = cargo.fullProjectDescription(childContext.project, projectDirectory, object : ProcessAdapter() {
+                override fun startNotified(event: ProcessEvent) {
+                    val processHandler = event.processHandler as? BaseProcessHandler<*> ?: return
+                    val commandLine = processHandler.commandLine
+                    if (" metadata " in commandLine) {
+                        CargoEventService.getInstance(childContext.project).onMetadataCall(projectDirectory)
+                    }
+                }
```

Comment:
Why `onMetadataCall` call is in `startNotified` but not just before `cargo.fullProjectDescription`? If it is reasonable, let's add a comment about it

Reply:
Not, it's not reasonable. Thanks for the suggestion!